ttnn/cpp/ttnn/operations/moreh/moreh_sum_backward/moreh_sum_backward.cpp

```diff
@@ -0,0 +1,32 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "moreh_sum_backward.hpp"
+
+#include "device/moreh_sum_backward_device_operation.hpp"
+#include "ttnn/deprecated/tt_dnn/op_library/moreh_helper_functions.hpp"
+
+namespace ttnn::operations::moreh::moreh_sum_backward {
+Tensor MorehSumBackward::invoke(
+    const Tensor& output_grad,
+    const std::optional<Tensor>& input,
+    std::optional<std::variant<int64_t, std::vector<int64_t>>> dim,
+    bool keepdim,
+    const std::optional<Tensor>& input_grad,
+    const std::optional<MemoryConfig>& input_grad_mem_config,
+    const std::optional<DeviceComputeKernelConfig>& compute_kernel_config) {
+    TT_FATAL((input.has_value() || input_grad.has_value()), "either input or input_grad must have a value");
+    uint32_t rank = input.has_value() ? input->get_shape().value.rank() : input_grad->get_shape().value.rank();
+    std::vector<int64_t> dims = tt::operations::primary::get_dim(dim, rank);
+    std::sort(dims.begin(), dims.end());
+    return ttnn::prim::moreh_sum_backward(
+        output_grad,
+        input,
+        dims,
+        keepdim,
+        input_grad,
+        input_grad_mem_config.value_or(operation::DEFAULT_OUTPUT_MEMORY_CONFIG),
```

Comment:
```suggestion
        input_grad_mem_config.value_or(output_grad.memory_config()),
```
Using the default memory config can cause incorrect behavior where DRAM output is produced when L1 input is provided. In such cases, the memory config of the input should be used.